src/core/root.js

```diff
@@ -155,3 +147,22 @@ export function closeEdge(): void {
   for (const context of allContexts) context.close().catch(() => {})
   allContexts = []
 }
+
+/**
+ * We only accept *.edge.app or localhost as valid domain names.
+ */
+export function validateServer(server: string): void {
+  const url = new URL(server)
+
+  if (url.protocol === 'http:') {
+    if (url.hostname === 'localhost') return
+  }
+  if (url.protocol === 'https:') {
+    if (url.hostname === 'localhost') return
+    if (/^([A-Za-z0-9_-]+\.)*edge\.app$/.test(url.hostname)) return
```

Comment:
needs to also support edgetest.app

Reply:
I put my rationale in both the commit message and the pull request. We have never used edgetest for anything, so I would like to remove it until we have a real reason to add it back. We consider our login server to be sensitive infrastructure (one of the few repos we don't make open source), so it will always be deployed `*.edge.app`, such as `login-test.edge.app`.